rust_src/src/lists.rs

```diff
@@ -98,19 +98,13 @@ fn cdr_safe(object: LispObject) -> LispObject {
 #[lisp_fn]
 pub fn nthcdr(n: LispObject, list: LispObject) -> LispObject {
     let num = n.as_fixnum_or_error();
-    let mut tail = list;
+    let mut it = list.iter_tails();
```

Comment:
The whole rest of the function is now just `it.nth(num)`?

Reply:
Sounds right